transports/onion/src/lib.rs

```diff
@@ -0,0 +1,232 @@
+            phantom: PhantomData::default(),
+        })
+    }
+
+    pub async fn bootstrap(&self) -> Result<(), OnionError> {
+        self.client.bootstrap().await
+    }
+
+    pub fn with_address_conversion(&mut self, conversion_mode: AddressConversion) -> &mut Self {
+        self.conversion_mode = conversion_mode;
+        self
+    }
+}
+
+macro_rules! default_constructor {
+    () => {
+        pub async fn bootstrapped() -> Result<Self, OnionError> {
+            let builder = Self::builder();
+            let ret = Self::from_builder(builder, AddressConversion::DnsOnly)?;
+            ret.bootstrap().await?;
+            Ok(ret)
+        }
+    };
+}
+
+#[cfg(all(feature = "native-tls", feature = "async-std"))]
+#[cfg_attr(docsrs, doc(cfg(all(feature = "native-tls", feature = "async-std"))))]
+impl<S> OnionTransport<tor_rtcompat::async_std::AsyncStdNativeTlsRuntime, S> {
+    pub fn builder() -> OnionBuilder<tor_rtcompat::async_std::AsyncStdNativeTlsRuntime> {
+        let runtime = tor_rtcompat::async_std::AsyncStdNativeTlsRuntime::current()
+            .expect("Couldn't get the current async_std native-tls runtime");
+        TorClient::with_runtime(runtime)
+    }
+    default_constructor!();
+}
+
+#[cfg(all(feature = "rustls", feature = "async-std"))]
+#[cfg_attr(docsrs, doc(cfg(all(feature = "rustls", feature = "async-std"))))]
+impl<S> OnionTransport<tor_rtcompat::async_std::AsyncStdRustlsRuntime, S> {
+    pub fn builder() -> OnionBuilder<tor_rtcompat::async_std::AsyncStdRustlsRuntime> {
+        let runtime = tor_rtcompat::async_std::AsyncStdRustlsRuntime::current()
+            .expect("Couldn't get the current async_std rustls runtime");
+        TorClient::with_runtime(runtime)
+    }
+    default_constructor!();
+}
+
+#[cfg(all(feature = "native-tls", feature = "tokio"))]
+#[cfg_attr(docsrs, doc(cfg(all(feature = "native-tls", feature = "tokio"))))]
+impl<S> OnionTransport<tor_rtcompat::tokio::TokioNativeTlsRuntime, S> {
+    pub fn builder() -> OnionBuilder<tor_rtcompat::tokio::TokioNativeTlsRuntime> {
+        let runtime = tor_rtcompat::tokio::TokioNativeTlsRuntime::current()
+            .expect("Couldn't get the current tokio native-tls runtime");
+        TorClient::with_runtime(runtime)
+    }
+    default_constructor!();
+}
+
+#[cfg(all(feature = "rustls", feature = "tokio"))]
+#[cfg_attr(docsrs, doc(cfg(all(feature = "rustls", feature = "tokio"))))]
+impl<S> OnionTransport<tor_rtcompat::tokio::TokioRustlsRuntime, S> {
+    pub fn builder() -> OnionBuilder<tor_rtcompat::tokio::TokioRustlsRuntime> {
+        let runtime = tor_rtcompat::tokio::TokioRustlsRuntime::current()
+            .expect("Couldn't get the current tokio rustls runtime");
+        TorClient::with_runtime(runtime)
+    }
+    default_constructor!();
+}
+
+#[cfg(all(feature = "native-tls", feature = "async-std"))]
+#[cfg_attr(docsrs, doc(cfg(all(feature = "native-tls", feature = "async-std"))))]
+pub type OnionAsyncStdNativeTlsTransport =
```

Comment:
Nit: I find it a bit odd that we are splitting `Onion` and `Transport` apart here. How about:

```suggestion
pub type AsyncStdNativeTlsOnionTransport =
```